File: 로또의 최고 순위와 최저 순위.py
# 일치하는 숫자의 개수를 넣으면 알맞은 등수를 반환한다.
def func(num):
    if num == 6:
        return 1
    elif num == 5:
        return 2
    elif num == 4:
        return 3
    elif num == 3:
        return 4
    elif num == 2:
        return 5
    elif num <= 1:
        return 6


def solution(lottos, win_nums):
    answer = []
    zero_num = 0
    col_num = 0

    for i in lottos:
        if i == 0:
            zero_num += 1
        if i in win_nums:
            col_num += 1
    
    answer.append(func(col_num + zero_num))
    answer.append(func(col_num))

    return answer

File: 로또의 최고 순위와 최저 순위.java
import java.util.Arrays;

public class Solution {
	public boolean func(int n, int[] arr) {
		for (int i = 0; i < arr.length; i++) {
			if (n == arr[i]) {
				return true;
			}
		}

		return false;
	}

	public int[] solution(int[] lottos, int[] win_nums) {
		int[] answer = { 0, 0 };
		int correct_num = 0;
		int zero_num = 0;

		for (int i = 0; i < lottos.length; i++) {
			if (lottos[i] == 0) {
				zero_num += 1;
				continue;
			}

			if (func(lottos[i], win_nums)) {
				correct_num += 1;
			}
		}

		answer[0] = 7 - (correct_num + zero_num);
		answer[1] = 7 - correct_num;
		
		if (answer[0] > 6) {
			answer[0] = 6;
		}
		
		if (answer[1] > 6) {
			answer[1] = 6;
		}

		return answer;
	}
}